### ISSUE TITLE
Impossible to input certain hour values for certain dates using keyboard entry

### ISSUE THREAD

**Issue by craigsmitham** [telerik/kendo-angular#1366]
## I'm submitting a.

* Bug report 

*For certain dates, when using keyboard entry to modify the hour to a time greater than 20 (using HH format), the date input will not update.*

## Current behavior
1. Set the date input to any of the problematic dates below
2. Enter value greater than "20" in the hour segment of the input
3. *The entered value does not update the hour segment of the input*

## Expected behavior
1. Set the date input to any of the problematic dates below
2. Enter value greater than "20" in the hour segment of the input
3. *The entered value should appear in the hour segment of the input*

### Problematic Dates
* 03/11/2018
* 03/12/2017
* 03/10/2019
* 03/13/2016
* _Likely many others, as there appears to be a pattern here ..._

## Minimal reproduction of the problem with instructions
* Follow above steps with
* Plunker demo: https://plnkr.co/edit/sXyi8UKgFoovALez42CE?p=preview
* StackBlitz - https://stackblitz.com/edit/angular-hhzcnh?file=src%2Fapp%2Fapp.component.ts
* Demo Site: https://www.telerik.com/kendo-angular-ui/components/dateinputs/dateinput/

## Environment

Package versions:
N/A - reproducible on demo site

Browser:
- Chrome (desktop)

**Comment by ggkrustev** [comment]
Hi @craigsmitham, thanks for the detailed report. I tried to replicate the issue locally, but to no avail.

https://www.screencast.com/t/BivlBkj10l

The issue could be reproducible only in particular browser timezone. Could you share the TZ of the browser where you are testing? Also am I missing something applying the repro steps?

**Comment by craigsmitham** [comment]
@ggkrustev we've been able to reproduce this in both US central and pacific time zones. 

We now think the issue has something to do with daylight savings. Any date where daylight savings come into effect this will be an issue.

**Comment by ggkrustev** [comment]
@craigsmitham, thanks for the clarification. 

Indeed, the component will not accept the first typed character - `2`, because this will produce a non-existing date - `03/11/2018 02:00`. Basically, when a `2` is typed, the component tries to create the aforementioned date, browser adjust the time according DST info and creates `03/11/2018 03:00`. Hence any additional character will produce invalid `hour` (> 24) and will be ignored.

Although the described behavior is somewhat expected, the UX is not great. We will further investigate the case and will post updates while research progresses.

**Comment by craigsmitham** [comment]
THe HTML5 datetime-local input does not appear to have this issue: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local

**Comment by ggkrustev** [comment]
Yep, I checked with the `datetime-local` control and noticed that it behaves correctly. As I mentioned, we recognize the defect and will research how to improve current behavior.
Stay tuned for updates.

**Comment by PlasticLizard** [comment]
Running into this again this year - it's quite surprising for our users - any progress? Thanks! - Nathan

**Comment by billm23** [comment]
It's been a couple years now, any progress on this yet?

**Comment by craigsmitham** [comment]
Any update?

**Comment by svetq** [comment]
It is not possible to enter Dates with their time being exactly daylight saving time for any specific time zone as such Dates with time don't exist. In reality, the time is either moved one hour forward or backward. We could indeed handle such cases by introducing a new behavior of the component. I will close this issue in favor of the following feature request item:
https://feedback.telerik.com/kendo-angular-ui/1679664-handle-dates-that-fall-on-daylight-saving-time

Feel free to add any additional comments, suggestions, and/or details to it.

**Comment by Miedziaq** [comment]
> It is not possible to enter Dates with their time being exactly daylight saving time for any specific time zone as such Dates with time don't exist. In reality, the time is either moved one hour forward or backward. We could indeed handle such cases by introducing a new behavior of the component. I will close this issue in favor of the following feature request item: https://feedback.telerik.com/kendo-angular-ui/1679664-handle-dates-that-fall-on-daylight-saving-time
> 
> Please cast your votes and add any additional comments, suggestions, and/or details in it.

I reported that bug in TimePicker that disallow to set 02, 2 and 20+ hour at 26/03/2026. I got the same response.
But You are absolutely not right, Time is going forward **ALWAYS** at the last march saturday/sunday night. At 26.03 it was in 2023...
The next thing is that the time picker that shouldn be aware of DATE is infected by the same and WRONG logic.

even if there is some reason, we cannot type 20+ hour too in 24h system, and timepicker should not see on date
